libs/hwdrivers/src/rplidar/src/rplidar_driver.cpp

```diff
@@ -349,10 +570,7 @@ u_result RPlidarDriverSerialImpl::ascendScanData(rplidar_response_measurement_no
     }
 
     // all the data is invalid
-    if (i == count){
-        delete[] tmpbuffer;
-        return RESULT_OPERATION_FAIL;
-    }
+    if (i == count) return RESULT_OPERATION_FAIL;
 
     //Tune tail
     for (i = count - 1; i >= 0; i--) {
```

Comment:
size_t is unsigned, so this check isn't valid anymore.
Consider
for(rplidar_response_measurement_node_t * ptr = nodebuffer + count - 1; ptr != nodebuffer; ptr--)